src/CleanSoftwareCron.php

```diff
@@ -93,7 +93,7 @@ public static function run(?int $max): int
         $total += self::deleteItems(
             self::getSoftwareWithNoVersionsCriteria(),
             new Software(),
-            $max - $total
+            $max
```

Comment:
Why did you remove this ?
If call on line 86 already removed the X elements, then this call should not remove more than `$max - X` element.

Reply:
Functionally speaking, we want to delete a maximum number of version (ex: 200) and a maximum number of software (ex: 200). If I indicate "200", it must apply as is for both requests. Without this change, there might even be cases where software are never deleted.

Reply:
Side note, users will see 400 deleted items in the action's logs despite their configuration only allowing 200 and might get confused.

Reply:
Maybe handling it in 2 queries does not make sense then. You may delete some `SoftwareVersion` and then delete some `Software` that are not even parents of deleted `SoftwareVersion`.
`getSoftwareWithNoVersionsCriteria()` should probably also return softwares that have none of their version installed, and so the deletion could be made in a unique step.